src/Domain/ComputationDomainMetric.hpp

```diff
@@ -0,0 +1,87 @@
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace domain {
+
+/*!
+ * \brief The computation domain metric \f$\sum_k \frac{\partial x^k}{\partial
+ * \xi^i} \frac{\partial x^k}{\partial \xi^j}\f$, which is the flat spatial
+ * metric in element-logical coordinates.
+ *
+ * We define the "computation domain metric" to be the inner product of the i-th
+ * basis vector with the j-th basis vector, where the basis vectors are
+ * \f$\partial_\xi\f$, \f$\partial_\eta\f$, and \f$\partial_\zeta\f$. They can
+ * be expressed in the \f$x\f$-coordinate system, which gives a column of the
+ * Jacobian. Multiplying with the transpose from the left gives a matrix of the
+ * inner products of these basis vectors - the computation domain metric. It is
+ * just the spatial metric with respect to the logical coordinates, and we're
+ * assuming a flat metric here for now.
+ *
+ * When we need it, the computation domain metric in curved space can be
+ * computed just by a coordinate transformation of the spatial metric to logical
+ * coordinates, e.g. using `::transform::to_different_frame`.
+ */
+template <size_t Dim>
+void computation_domain_metric(
+    const gsl::not_null<tnsr::ii<DataVector, Dim, Frame::ElementLogical>*>
+        result,
+    const Jacobian<DataVector, Dim, Frame::ElementLogical, Frame::Inertial>&
+        jacobian) {
+  destructive_resize_components(result, jacobian.begin()->size());
+  for (size_t i = 0; i < Dim; ++i) {
+    for (size_t j = 0; j <= i; ++j) {
+      result->get(i, j) = 0.;
+      for (size_t k = 0; k < Dim; ++k) {
+        result->get(i, j) += jacobian.get(k, i) * jacobian.get(k, j);
+      }
+    }
+  }
+}
+
+namespace Tags {
+/// The computation domain metric.
+///
+/// \see domain::computation_domain_metric
+template <size_t Dim>
+struct ComputationDomainMetric : db::SimpleTag {
+  using type = tnsr::ii<DataVector, Dim, Frame::ElementLogical>;
+};
+
+/// Compute the flat-space computation domain metric from the inverse Jacobian.
+///
+/// \see domain::computation_domain_metric
+template <size_t Dim>
+struct ComputationDomainMetricCompute : ComputationDomainMetric<Dim>,
+                                        db::ComputeTag {
+  using base = ComputationDomainMetric<Dim>;
+  using return_type = typename base::type;
+  using argument_tags =
+      tmpl::list<domain::Tags::InverseJacobian<Dim, Frame::ElementLogical,
+                                               Frame::Inertial>>;
+  static void function(
+      const gsl::not_null<tnsr::ii<DataVector, Dim, Frame::ElementLogical>*>
+          result,
+      const ::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
+                              Frame::Inertial>& inv_jacobian) {
+    // We have the inverse Jacobian available, so just do a numerical inverse to
```

Comment:
Comment: Is the Jacobian not available? In the CoordinateMaps the Jacobian is computed analytically and the inverse Jacobian is computed by computing the Jacobian first, then inverting numerically, so if I'm understanding correctly we would be doing two numerical inversions for a quantity we have analytically. (Unless this is a detail of the time dependent maps that I'm missing, which is possible!)

Question: I wasn't expecting to be able to visualize the logical metric for time-dependent maps so soon, is that possible here? If so, that's great!

Reply:
For time-dep maps the element-logical to inertial Jacobian isn't currently in the DataBox. If it makes its way in there at some point then we can replace it here, as the comment says.